File: js/euler57.js
function euler57(){
    var num = 1n;
    var den = 1n;
    var res = 0;
	// 1/1, 3/2, 7/5, 17/12, 41/29, ...
    for (let i = 0n; i < 1000n; i++){
        let newNum = 2n * den + num;
        let newDen = den + num;
        num = newNum;
        den = newDen;
		// numerator has more digits than denominator
        if (num.toString().length > den.toString().length) res++;
    }
    return res;
}

console.log(euler57());

File: go/euler57.go
package main
import (
    "fmt"
    "math/big"
)

func main() {
    res := 0
    num := big.NewInt(3)
    den := big.NewInt(2)
    two := big.NewInt(2)
    for i := 1; i <= 1000; i++ {
        newNum := new(big.Int).Mul(den, two)
        newNum.Add(newNum, num)
        newDen := new(big.Int).Add(num, den)
        
        if len(newNum.String()) > len(newDen.String()) {
            res++
        }
        num = newNum
        den = newDen
    }
    fmt.Println(res)
}
